chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
+The Rust and Haskell integrations for instance contain at best only a fraction of code in their respective language.
+
+| Plugin                    | lines of code                                                |
+| ------------------------- | ------------------------------------------------------------ |
+| intellij-haskell          | 17249 (Java) + 13476 (Scala) **+ 0 (Haskell)**               |
+| intellij-rust             | 229131 (Kotlin) + 3958 (Rust)                                |
+| intellij-scala            | 39382 (Java) + 478904 (Scala)                                |
+| intellij-kotlin           | 182372 (Java) + 563394 (Kotlin)                              |
+| intellij-community/python | 47720 (C) + 248177 (Java) + 37101 (Kotlin) + 277125 (Python) |
+
+: Comparison of the size for different IntelliJ platform plugins {#tbl:plugins-size}
+
+Naturally, development efforts at this size would gravitate around the most promising solution, stifling the progress on competing platforms [@intellij-comparison-eclipse].
+Additionally, it would lock-in programmers into a specific platform for its language support regardless of their personal preference.
+
+### IDE Abstraction
+
+#### Monto
+
+The authors of the Monto project[@monto,@monto-disintegrated] call this the "IDE Portability Problem".
+They compare the situation with the task of compiling different high level languages to a set of CPU architectures.
+The answer to that problem was an intermediate representation (IR).
+Compilers could transform input languages into this IR and in turn generate assembly for different architectures from a single input format.
+
+With Monto, Kreidel et al propose a similar idea for IDE portability.
+The paper describes the *Monto IR* and how they use a *Message Broker* to receive events from the Editor and dispatch them to *Monto Services*.
+
+The Monto IR is a language agnostic and editor independent, JSON serialized tree-like model.
+Additionally, the IR maintains low level syntax highlighting information (font, color, style, etc.) but leaves the highlighting to the language specific service.
+
+The processing and modification of the source code and IR is performed by *Monto Services*.
+Services implement specific actions, e.g. parsing, outlining or highlighting.
+A central broker connects the services with each other and the editor.
+
+Since Monto performs all work on the IR, independent of the editor, and serializes the IR as JSON messages, the language used to implement *Monto Services* can be chosen freely giving even more flexibility.
+
+The Editor extension's responsibility is to act as a source and sink for data.
+It sends Monto compliant messages to the broker and receives processing results such as (error) reports.
+The communication is based on the ZeroMQ[zeromq] technology which was chosen because it is lightweight and available in manly languages [@monto-disintegrated] allowing to make use of existing language tools.
+
```

Comment:
This Monto stuff is really interesting.

Reply:
Yes it looks really flexible, but burdened itself with doing basically everything including lowest level syntax coloring.